test/integration/certificates/util.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificates
+
+import (
+	"context"
+	"encoding/pem"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+	"github.com/jetstack/cert-manager/pkg/util/pki"
+)
+
+// ensureConditionNotApplied ensures that the provided condition is *not* applied to the certificate within 2 seconds.
+func ensureConditionNotApplied(ctx context.Context, t *testing.T, cmCl cmclient.Interface, cert *cmapi.Certificate, condition cmapi.CertificateCondition) {
+	err := wait.Poll(time.Millisecond*200, time.Second*2, func() (done bool, err error) {
```

Comment:
With custom test helper functions, it may be beneficial to call `t.Helper` on the first line of each. This way, in the test failure output, we see the line number in the actual test that calls `ensureConditionNotApplied`, rather than the line number in this `util.go` file.

Reply:
Thanks for the suggestion- tried it out and it definitely makes it easier to follow where an error occured!

Reply:
woaahhh I wasn't aware of `t.Helper`!!! 🤯
I wish I had known that before 😅

Reply:
^ good reminder that I need to get back and clean up this PR 😄